### PR TITLE
PRIME-956 Remote User Email

### PR DESCRIPTION
**Pull request by AnaisHebert**

### CODE REVIEW

File: prime-dotnet-webapi/Services/EmailService.cs [hunk 1]
```diff
@@ -165,6 +170,20 @@ public async Task SendRemoteUsersUpdatedAsync(Site site)
             await Send(PRIME_EMAIL, new[] { MOH_EMAIL, PRIME_SUPPORT_EMAIL }, subject, body, attachments);
         }
 
+        public async Task SendRemoteUsersNotificationAsync(Site site, IEnumerable<RemoteUser> remoteUsers)
+        {
+            var subject = "Remote Practitionner Notification";
+
+            foreach (var remoteUser in remoteUsers)
+            {
+                var body = await _razorConverterService.RenderViewToStringAsync(
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EmailService.cs:179]: Doesnt look like we need to render the body each time, we can render it once and use it for each email.

File: prime-dotnet-webapi/Services/EmailService.cs [hunk 2]
```diff
@@ -165,6 +170,20 @@ public async Task SendRemoteUsersUpdatedAsync(Site site)
             await Send(PRIME_EMAIL, new[] { MOH_EMAIL, PRIME_SUPPORT_EMAIL }, subject, body, attachments);
         }
 
+        public async Task SendRemoteUsersNotificationAsync(Site site, IEnumerable<RemoteUser> remoteUsers)
+        {
+            var subject = "Remote Practitionner Notification";
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/EmailService.cs:175]: "Practitioner"

File: prime-dotnet-webapi/Views/Emails/RemoteUserNotificationEmail.cshtml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+}
+
+The Ministry of Health has been notified that you require Remote Access at
+<br>
+Organization Name: @site.Organization.Name
+<br>
+Site Address: @site.PhysicalAddress.Street, @site.PhysicalAddress.City
+<br>
+<br>
+To complete your approval for Remote Access, please ensure you have indicated you require Remote Access on our profile at
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Views/Emails/RemoteUserNotificationEmail.cshtml:15]: require Remote Access on "your" profile

File: prime-dotnet-webapi/Views/Emails/RemoteUserNotificationEmail.cshtml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+<a href="https://pharmanetenrolment.gov.bc.ca/info">https://pharmanetenrolment.gov.bc.ca/info</a>.
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Views/Emails/RemoteUserNotificationEmail.cshtml:16]: Might as well use the FRONTEND_URL environment variable here

File: prime-dotnet-webapi/Controllers/SitesController.cs [hunk 5]
```diff
@@ -385,5 +386,55 @@ public async Task<ActionResult> sendRemoteUsersEmail(int siteId)
             return NoContent();
         }
 
+        // POST: api/Sites/5/remote-users-email-admin
+        /// <summary>
+        /// Send HIBC an email when remote users are updated for a submitted site
+        /// </summary>
+        /// <param name="siteId"></param>
+        [HttpPost("{siteId}/remote-users-email-admin", Name = nameof(sendRemoteUsersEmailAdmin))]
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Controllers/SitesController.cs:394]: Can only admin send emails using these two endpoints? If so, we should add the admin auth header to these.

File: prime-dotnet-webapi/Services/EmailService.cs (hunk 2 above)
Comment (AnaisHebert) [reply prime-dotnet-webapi/Services/EmailService.cs:175]: image
Thats how it is spelled in the ticket, not sure if intentionally.

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 5 above)
Comment (AnaisHebert) [reply prime-dotnet-webapi/Controllers/SitesController.cs:394]: No it just sends an email to the admins

File: prime-dotnet-webapi/Services/EmailService.cs (hunk 2 above)
Comment (james-hollinger) [reply prime-dotnet-webapi/Services/EmailService.cs:175]: oh dear 😅

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 5 above)
Comment (AnaisHebert) [reply prime-dotnet-webapi/Controllers/SitesController.cs:394]: And this new one sends an email to the remote user

File: prime-dotnet-webapi/Services/EmailService.cs (hunk 2 above)
Comment (mtpultz) [reply prime-dotnet-webapi/Services/EmailService.cs:175]: Confirmed definitely not spelled correctly

File: prime-dotnet-webapi/Controllers/SitesController.cs (hunk 5 above)
Comment (james-hollinger) [reply prime-dotnet-webapi/Controllers/SitesController.cs:394]: As is, these two endpoints can be triggered by any user to send emails to other people. Can we restrict these two endpoints in any way, similar to what we are doing on other endpoints?

**Review by mtpultz — APPROVED** [review]
image

**Pull request merged**